File: app/src/main/java/rod/bailey/trafficatnsw/cameras/FavouriteCameraDialogPresenter.java
package rod.bailey.trafficatnsw.cameras;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;

import rod.bailey.trafficatnsw.util.MLog;

public class FavouriteCameraDialogPresenter {

	private static final String TAG = FavouriteCameraDialogPresenter.class
			.getSimpleName();

	/** Listens for button taps on the radio buttons */
	private final class AlertsDialogListener implements
			DialogInterface.OnClickListener {

		@Override
		public void onClick(DialogInterface arg0, int index) {

			switch (index) {
			case DialogInterface.BUTTON_POSITIVE:
				MLog.INSTANCE.i(TAG, "*** About to toggle favourite state of camera " + camera.getIndex());
				camera.setFavourite(!camera.isFavourite());
				break;

			case DialogInterface.BUTTON_NEGATIVE:
				break;
			}
		}
	}

	private final TrafficCamera camera;

	public FavouriteCameraDialogPresenter(TrafficCamera camera) {
		assert camera != null;
		this.camera = camera;
		MLog.INSTANCE.i(TAG, "Presenter created for a camera whose favourite status is "
				+ camera.isFavourite());
	}

	public AlertDialog build(Context ctx) {
		assert ctx != null;

		AlertsDialogListener listener = new AlertsDialogListener();

		AlertDialog.Builder builder = new AlertDialog.Builder(ctx);
		builder.setTitle("Favourite");
		builder.setPositiveButton(
				camera.isFavourite() ? "Remove from Favourites"
						: "Make a Favourite", listener);
		builder.setNegativeButton("No", listener);
		builder.setMessage(camera.isFavourite() ? "Do you want to remove this camera from your list of favourite cameras?"
				: "Do you want to add this camera to your list of favourite cameras?");

		AlertDialog alert = builder.create();
		alert.setCanceledOnTouchOutside(true);

		return alert;
	}
}

File: app/src/main/java/rod/bailey/trafficatnsw/cameras/FavouriteCameraDialogPresenter.kt
package rod.bailey.trafficatnsw.cameras

import android.app.AlertDialog
import android.content.Context
import android.content.DialogInterface
import rod.bailey.trafficatnsw.util.MLog

class FavouriteCameraDialogPresenter(private val camera: TrafficCamera) {
	/** Listens for button taps on the radio buttons  */
	private inner class AlertsDialogListener : DialogInterface.OnClickListener {
		override fun onClick(arg0: DialogInterface, index: Int) {
			when (index) {
				DialogInterface.BUTTON_POSITIVE -> {
					MLog.i(TAG,
						   "*** About to toggle favourite state of camera " + camera.index)
					camera.isFavourite = !camera.isFavourite
				}
				DialogInterface.BUTTON_NEGATIVE -> {
				}
			}
		}
	}

	init {
		MLog.i(TAG,
			   "Presenter created for a camera whose favourite status is " + camera.isFavourite)
	}

	fun build(ctx: Context?): AlertDialog {
		val listener = AlertsDialogListener()
		val builder = AlertDialog.Builder(ctx)
		builder.setTitle("Favourite")
		builder.setPositiveButton(
			if (camera.isFavourite)
				"Remove from Favourites"
			else
				"Make a Favourite", listener)
		builder.setNegativeButton("No", listener)
		builder.setMessage(if (camera.isFavourite)
							   "Do you want to remove this camera from your list of favourite cameras?"
						   else
							   "Do you want to add this camera to your list of favourite cameras?")
		val alert = builder.create()
		alert.setCanceledOnTouchOutside(true)

		return alert
	}

	companion object {
		private val TAG = FavouriteCameraDialogPresenter::class.java
			.simpleName
	}
}
